feat: Create Shopping Page #164

Agbeleshe opened the pull request:
…eria

# 📝 Pull Request Title
Completed Shopping Page

## 🛠️ Issue
- Closes #164 

## 📖 Description
- I created the shopping page according to the "Acceptance Criteria"

## ✅ Changes made

- The page is accessible at /shopping. ✔
- The UI match the reference image.✔
- Implemented a tab-based filtering system with the specified options.✔
- Implement a search bar to filter products by name.✔
- Displayed a table with the specified columns.✔
- Showed the total number of purchases at the bottom of the table.✔
- Ensured responsiveness for both desktop and mobile views.✔
- Updated the en.ts and es.ts in /locales to include translations of the page✔
- Used shadcn/ui components for Table, Button, Badge, and Input.✔
- Used Lucide React Icons.✔
- Used mock data for initial development. It will be replaced with API integration later. ✔

18. ## 🖼️ Media (screenshots/videos)

![image](https://github.com/user-attachments/assets/0f0220a3-4ea1-4106-b4c1-65875cc9e9fc)

-

## 📜 Additional Notes
Thank you :)
-

derianrddev commented on apps/frontend/app/(routes)/shopping/shopping-table.tsx:
```diff
@@ -0,0 +1,431 @@
+	TooltipProvider,
+	TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { AlertCircle, CheckCircle2, Clock, Eye, Search, X } from "lucide-react";
```
Remove `X` icon is not used

derianrddev commented on apps/frontend/app/(routes)/shopping/shopping-table.tsx:
The translations for this component have already been added in `en.ts` and `es.ts`, but they are not yet applied in the code. Please replace the hardcoded text with `useTranslations` to ensure proper localization.

derianrddev commented on apps/frontend/app/(routes)/shopping/shopping-table.tsx:
```diff
@@ -0,0 +1,431 @@
+const ShoppingTable = () => {
+	const [activeTab, setActiveTab] = useState<TabType>("All");
+	const [searchTerm, setSearchTerm] = useState("");
+	const [data] = useState<ShoppingData[]>(initialData);
```
The `data` state is initialized with `initialData`, but I don't see it being updated anywhere. If it doesn’t need to change dynamically, you could remove the state and use `initialData` directly in the filter function.

derianrddev reviewed (changes requested):
@Agbeleshe 
Please add the title 'My Shopping' to the page and fix the linter errors. You can check them by running `npm run lint:frontend` from the monorepo root or `npm run lint` if you're inside the `frontend` folder.

Make sure to address the comments below as well.

![image](https://github.com/user-attachments/assets/74ce4673-8da6-4d23-aa04-67a7023b5581)

derianrddev reviewed (approved):
@Agbeleshe LGTM! wait for the other reviews.

Agbeleshe commented:
Hello @derianrddev  
Patiently waiting for second review

diegoTech14 commented:
@Agbeleshe Hi sir, could you address the followind points please?

1. We need that the app supports tablet responsiveness

2. When we activate dark mode, the card and the search bar do not change color on mobile and on desktop the search bar does not change either.
![image](https://github.com/user-attachments/assets/2d787c9c-e9d0-4c0a-836e-d2239bf8c013)

Desktop:
![image](https://github.com/user-attachments/assets/c33a898f-a136-4c24-9343-48fb266dc999)


3. In the order details we can not see the status, just appear a button:
![image](https://github.com/user-attachments/assets/0ef15da6-1a5c-46c5-aa0a-6b9b1bcf8439)

Agbeleshe commented:
Okay noted

Agbeleshe commented:
@diego, Hello👋,

### Changes made 

**1. Tablet Responsiveness**:

- Added a three-tier responsive layout system (mobile, tablet, desktop) instead of just mobile/desktop
- For tablets, I've created a grid layout with 2 columns for card display
- Updated the media query detection with size breakpoints at 640px and 1024px
- Modified the responsive classes to better handle different screen sizes
![image](https://github.com/user-attachments/assets/c8cae4d4-335e-4a16-a838-75bb731346ef)



2. Dark Mode Issues

- Added dark: variants to all background, text, and border colors
- Fixed the Card component with dark:bg-gray-800
- Fixed the search input with dark:bg-gray-700 dark:text-gray-100
- Added dark mode styles to all text elements with appropriate contrast
- Fixed border colors in dark mode with dark:border-gray-700
- Updated status badges to have proper dark mode styling.
![image](https://github.com/user-attachments/assets/04814cb6-f8f4-49ff-99ec-7b63f18f8c43)



3. Order Details Status Display

- Fixed the status display in the OrderDetails component by ensuring the Badge component correctly uses the status color and icon
- Added dark mode support for the Dialog component with dark:bg-gray-800
- Added dark mode support for all text in the modal
![image](https://github.com/user-attachments/assets/be29b2a6-ac7d-40c6-9f33-d54b0479899f)


Patiently waiting for your feedback :)

diegoTech14 commented:
**Hi @Agbeleshe could you resolve the following issues please?**

1. There are some issues in the es.ts / en.ts files because there are 2 "shopping" objects and the app doesn't recognize the language

![image](https://github.com/user-attachments/assets/3ce16bd0-8ce2-4fc2-8f6a-43139069dda0)

2. I think you are not following the color palette, we don't use a gray color
3. The state in the modal still does not appear

Agbeleshe commented:
Hello @diegoTech14 , 🧙‍♂️
### **Done with the specified changes**

1. I Deleted the duplicate shoppig object now the language are displaying. English and Spanish writeups are properly displayed now.
![image](https://github.com/user-attachments/assets/2c150198-5a18-414e-b13b-a7027032f5ba)
![image](https://github.com/user-attachments/assets/06572af4-8892-42c8-ad39-5cf153625ac3)


2. followed the color palet and removed the gray background for dark-mode (although i feel the gray bg was cooler)
![image](https://github.com/user-attachments/assets/7093443a-256b-486c-bf87-3430100cf645)


3. Modal sate working for all now 
![image](https://github.com/user-attachments/assets/6198ea0b-c93e-4bde-ae5d-de26620bcf14)
![image](https://github.com/user-attachments/assets/59deea4e-0ce5-45f3-8c1c-15336eea1d6b)
![image](https://github.com/user-attachments/assets/cdaeb7bc-0e61-47da-b7a1-c2e8b77a4d93)
![image](https://github.com/user-attachments/assets/ceac93c0-dd7a-4c08-b09a-f6b8afce900b)



**Waiting patiently for your response thank you :)**

diegoTech14 reviewed (changes requested):
Hello @Agbeleshe PR looks good, just a minor change:

It seems that you removed important data from the translations, could you place them inside the `shopping` object please?

![image](https://github.com/user-attachments/assets/cbd7a5af-5afa-400c-a941-d7b05945d192)

![image](https://github.com/user-attachments/assets/5186e736-7292-4e87-94bf-2690de9e60a7)

Agbeleshe commented:
@diegoTech14 Hello ,
Done with the changes. 

Patiently waiting for your review :)
